server-utils/src/cors.rs

```diff
@@ -102,6 +102,9 @@ impl ops::Deref for Origin {
 
 /// Origins allowed to access
 #[derive(Debug, Clone, PartialEq, Eq)]
+
+/// 1) Adding new CORS protocol to enum below
```

Comment:
I think it should rather be a separate enum as outlined in https://github.com/paritytech/parity/issues/6616#issuecomment-337229845

If we want to be super strict the headers could be defined on the `Origin` level, but IMHO global setting for all allowed Origins are fine.

So the steps would be:
1. Add new enum (in that file most likely)
2. Wherever `get_cors_header` is used we should also compare the headers in the request with the allowed list. If that matches and we send CORS headers we should also add `Access-Control-Allowed-Headers` to the response.